### DISCUSSION TITLE
License clarification

### DISCUSSION THREAD

**Discussion by lucaskanashiro** [valkey-io/valkey#830] [Q&A]
Hi,

I am packaging valkey version 7.2.5 in Debian, and one question I got from a reviewer is about the license of the following files:

* https://github.com/valkey-io/valkey/blob/unstable/src/sha256.c
* https://github.com/valkey-io/valkey/blob/unstable/deps/hiredis/adapters/macosx.h
* https://github.com/valkey-io/valkey/blob/unstable/deps/hiredis/examples/example-macosx.c

The first one has an author and no copyright nor license info. The two others have `Copyright (c) 2015 Dmitry Bakhvalov. All rights reserved.`, the "All rights reserved" may not be considered 100% free license. Could upstream maintainers/developers clarify under which license those files are shipped?

TIA!

**Comment by hyandell** [comment]
Interested observer here.

On the sha256.c file, the stated 'license' is "This code is presented "as is" without any guarantees.', which I agree is just that little bit vague. The canonical source nowadays looks to be here: https://github.com/B-Con/crypto-algorithms/tree/master which states it's under public domain:

> This code is released into the public domain free of any restrictions. The author requests acknowledgement if the code is used, but does not require it. This code is provided free of any liability and without any quality claims by the author.

The difference between https://raw.githubusercontent.com/valkey-io/valkey/unstable/src/sha256.c and https://raw.githubusercontent.com/B-Con/crypto-algorithms/master/sha256.c seems both neglible, but also appears to be due to this commit: https://github.com/valkey-io/valkey/commit/b91d8b289bb64965c5eaa445809f9f49293e99c0#diff-d8a7b8c4e1c9f1a256cde1364a5b124adb14083e70181f85beeefa3d5a1debce
